### PR TITLE
Metadata signing via Rabbit MQ

### PR DESCRIPTION
**Pull request by dandlezzz**
This PR brings our metadata signing to async processing coordinated via Rabbit MQ.

Via Rabbit MQ we can defer signing NFT metadata and take it outside of the web request flow. This will help accommodate what can occasionally be a pretty long wait for the Solana transaction to confirm.

We have one queue for signing, if the signing fails it goes into a "dead-letter" queue. From the rabbit mq interface we can requeue jobs, multiple at once, but manually. Eventually, we can add automatic retries. 

 How to run the consumer `yarn run consumers:sign-metadata`.
This will spin up a consumer in the background. All signing taks will go to the queue from here on out.

### CODE REVIEW

**Review by kristianeboe — APPROVED** [review]
Clean code!

File: tasks/consumers/sign-metadata.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      } catch {
+        console.error('Invalid public keys');
+        channel.reject(msg, false)
+        return;
+      }
+
+      try {
+        const tx = new Transaction();
+
+        signMetadata(metadata, keypair.publicKey, tx, metaProgramId);
```
Comment (kespinola) [inline tasks/consumers/sign-metadata.ts:125]: The sdk has this action. Can we switch to that at this time?

File: src/modules/metadata-signing.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+
+export const signingQueue = "metadata-signging"
```
Comment (kespinola) [inline src/modules/metadata-signing.ts:2]: Move this to an environment variable?

File: tasks/consumers/sign-metadata.ts (hunk 1 above)
Comment (dandlezzz) [reply tasks/consumers/sign-metadata.ts:125]: I'd like to address that in a separate PR.

File: src/modules/metadata-signing.ts (hunk 2 above)
Comment (dandlezzz) [reply src/modules/metadata-signing.ts:2]: It's better if these don't change without a code update. The assertQueue method creates a queue if one doesn't exist by that name. Don't really want them changed too often.

**Pull request merged**